### PR TITLE
Don't rely on vendored tools

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
We now use `go install` to get go-md2man, ginkgo as well as the uber maintained version of gomock.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@cri-o/cri-o-maintainers PTAL

**Comment by haircommander** [comment]
/lgtm
nice cleanup!

File: hack/go-install.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -uo pipefail
+
+DEST_DIR=${1:-}
+CMD=${2:-}
+GO_INSTALL_LOC=${3:-}
+
+if [[ ("$DEST_DIR" == "" || "$CMD" == "" || "$GO_INSTALL_LOC" == "") ]]; then
```
Comment (kwilczynski) [inline hack/go-install.sh:8]: Curious, as I am not sure, do the brackets make any difference here?

File: hack/go-install.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    echo "Usage: $0 DEST_DIR CMD GO_INSTALL_LOC"
+    exit 1
+fi
+
+set +e
+CMD_PATH=$(command -v "$CMD")
+set -e
```
Comment (kwilczynski) [inline hack/go-install.sh:15]: What if we make it more explicit, for example:

```suggestion
CMD_PATH=$(command -v "$CMD" || true)
```

File: hack/go-install.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+if [ "$CMD_PATH" == "" ]; then
```
Comment (kwilczynski) [inline hack/go-install.sh:17]: For `[`, perhaps `-n` or `-z` would be better? For `[[` (the built-in), `==` is probably fine.

File: hack/go-install.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -uo pipefail
```
Comment (kwilczynski) [inline hack/go-install.sh:2]:
```suggestion
#!/usr/bin/env bash

set -uo pipefail
```

File: hack/go-install.sh (hunk 2 above)
Comment (saschagrunert) [reply hack/go-install.sh]: Yep adjusted as suggested.

File: hack/go-install.sh (hunk 1 above)
Comment (saschagrunert) [reply hack/go-install.sh]: I think we can remove them.

File: hack/go-install.sh (hunk 3 above)
Comment (saschagrunert) [reply hack/go-install.sh]: Using `-z` now :+1:

File: hack/go-install.sh (hunk 4 above)
Comment (saschagrunert) [reply hack/go-install.sh]: Fixed

**Comment by saschagrunert** [comment]
/override ci/prow/ci-e2e-evented-pleg


@cri-o/cri-o-maintainers PTAL

**Pull request merged**